benchmark-service/benchmark/benchmark.py

```diff
@@ -33,15 +96,50 @@ async def greet_user(request: web.Request) -> web.Response:
     return response
 
 
+@router.get('/name/{name}')
+async def show_name(request: web.Request) -> web.Response:
+    benchmarks = get_benchmarks(filepath)
+    name_data = next((item for item in benchmarks['data'] if item['name'] == str(request.match_info['name'])), None)
+    fig = px.scatter(x=[item for item in range(0, len(name_data['times']))], y=name_data['times'])
+
+    # context = {
+    #     'name': request.match_info.get('name', ''),
+    #     'name_data': name_data,
+    #     'fig': json.dumps(fig, cls=plotly.utils.PlotlyJSONEncoder)
+    # }
+    plot = json.dumps(fig, cls=plotly.utils.PlotlyJSONEncoder)
+    context = {
+        'name': request.match_info.get('name', ''),
+        'plot': plot
+    }
+
+    response = aiohttp_jinja2.render_template('user.html', request,
+                                              context=context)
+    return response
+
+
 @router.get('/')
 @router.get('')
 @web_authenticated_developers_only(redirect=False)
 async def index(request: web.Request, userdata) -> Dict[str, Any]:  # pylint: disable=unused-argument
     context = {
         'current_date': 'July 10, 2020'
     }
+    benchmarks = get_benchmarks(filepath)
     response = aiohttp_jinja2.render_template('index.html', request,
-                                              context=context)
+                                              context=benchmarks)
+    return response
+
+
+@router.post('/lookup')
+@web_authenticated_developers_only(redirect=False)
```

Comment:
Here's you'd want to add the decorator `@check_csrf_token`. Take a look at what this does in gear.